### ISSUE TITLE
Display exception when creating multiple SMARTS instances

### ISSUE THREAD

**Issue by zbzhu99** [huawei-noah/SMARTS#184]
I encounter the "Display is not found" exception when trying to create multiple SMARTS instances:

![截屏2020-11-19 10 19 23](https://user-images.githubusercontent.com/32090492/99612531-b8bbd580-2a50-11eb-953b-1372be532c25.png)

To reproduce:

```python
train_env = gym.make(
    "smarts.env:hiway-v0",
    scenarios=scenarios,
    agent_specs={AGENT_ID: agent_spec},
    headless=headless,
    visdom=False,
    timestep_sec=0.1,
    sumo_headless=True,
    seed=seed,
    # envision_record_data_replay_path="./data_replay",
)

test_env = gym.make(
    "smarts.env:hiway-v0",
    scenarios=scenarios,
    agent_specs={AGENT_ID: agent_spec},
    headless=headless,
    visdom=False,
    timestep_sec=0.1,
    sumo_headless=True,
    seed=seed,
    # envision_record_data_replay_path="./data_replay",
)
```

**Development Environment:**

OS: Ubuntu 16.04

Python: 3.7.4

SUMO: 1.7.0

SMARTS: 0.4.4

**Comment by Gamenot** [comment]
Hello @zbzhu99 , this is a known issue. SMARTS uses a few dependencies that have the requirement for a singular instance. In order to use another SMARTS instance in the same thread the previous needs to be destroyed, or run on another process.

There is an `example/multi_instance.py` that shows one way to be able use more than one smarts environment.

https://github.com/huawei-noah/SMARTS/blob/master/examples/multi_instance.py#L68-L162

**Comment by davidrusu** [comment]
@Gamenot We might be able to solve this problem using the suggestions in this comment here: https://github.com/panda3d/panda3d/issues/996#issuecomment-679916768

**Comment by Gamenot** [comment]
@davidrusu Yes, it seems like `option 1` from that list of suggestions is the way to go eventually. We are already recommending `option 3` via `ray` but stripping away the globals so that multiple instances can run in the same context would lower complexity for the users

**Comment by alek5k** [comment]
Hi there, I wrote the original github issue over at Panda3D. Just wondering if you'd be willing to share your implementation of the referenced `option 1` from my github issue on Panda3D? I ended up going with option 3, but my training time is suffering :\

**Comment by davidrusu** [comment]
@alek5k We'd be happy to share it once we've got around to tackling this. Currently we are also relying on multiprocessing to avoid multiple instances of Showbase.

If you beat us to it, please come back and share your findings.

**Comment by alek5k** [comment]
Another relevant issue with Panda3D that you might run into if you are planning on running many instances in linux:

https://github.com/panda3d/panda3d/issues/1064

**Comment by davidrusu** [comment]
@alek5k Yeah we did hit that one with one of our users as well

Maybe you can find something of use from the other configuration options we tuned here: 

https://github.com/huawei-noah/SMARTS/blob/master/smarts/core/smarts.py#L62-L68

**Comment by Gamenot** [comment]
Since we are just using `Showbase` for rendering another option figured out here for us in particular is to use a `Showbase` subclass as a singleton.

**Comment by Gamenot** [comment]
This particular problem has been addressed using `option 2`. The rest of the issue is covered by #743 .
